pkg/api/v1/resource/helpers.go

```diff
@@ -83,20 +83,38 @@ func PodRequests(pod *v1.Pod, opts PodResourcesOptions) v1.ResourceList {
 		addResourceList(reqs, containerReqs)
 	}
 
+	restartableInitContainerReqs := v1.ResourceList{}
+	initContainerReqs := v1.ResourceList{}
 	// init containers define the minimum of any resource
 	// Note: In-place resize is not allowed for InitContainers, so no need to check for ResizeStatus value
 	for _, container := range pod.Spec.InitContainers {
-		containerReqs := container.Resources.Requests
+		containerReqs := container.Resources.Requests.DeepCopy()
```

Comment:
Mutating container requests is a bit wierd, and this helper can be used in somewhat performance intensive code.  I'd prefer not to allocate copies unless absolutely necessary - is there a way to write this that does not perform this DeepCopy or the one above, and avoids additional creations of heap objects?

Reply:
@tzneal 
It looks safe not to copy this here.
I'll revert this change. Let me know if I am wrong.

Reply:
LGTM, at one point I think `applyNonMissing` modified its parameter.